libnmstate/ifaces/ifaces.py

```diff
@@ -198,6 +193,31 @@ def _apply_copy_mac_from(self):
             else:
                 iface.apply_copy_mac_from(port_iface.mac)
 
+    def _validate_copy_mac_from_iface_is_in_port(self, iface):
+        iface_port = self._port_for_interface(iface)
+        if iface_port is None:
+            raise NmstateValueError(
+                f"The interface {iface.name} is holding invalid "
+                f"{Interface.COPY_MAC_FROM} property "
+                f"as the interface is not attached to any bridge"
+            )
+        if iface.copy_mac_from not in iface_port:
+            raise NmstateValueError(
+                f"The interface {iface.name} is holding invalid "
+                f"{Interface.COPY_MAC_FROM} property "
+                f"as {iface.copy_mac_from} is not in the port "
+                f"list: {iface.port}"
+            )
+
+    def _port_for_interface(self, iface):
+        if iface.type == InterfaceType.OVS_INTERFACE:
```

Comment:
This does not seem correct. It returns the bridge itself, not the list of its ports, no?

So while it would perform the check as expected, it does not do what the function name suggests.

Reply:
Resolved